frontend/lib/badge-url.js

```diff
@@ -12,13 +16,9 @@ export default function resolveBadgeUrl(url, baseUrl, options) {
   return resolveUrl(url, baseUrl, outQueryParams)
 }
 
-export function encodeField(s) {
-  return encodeURIComponent(s.replace(/-/g, '--').replace(/_/g, '__'))
-}
-
-export function staticBadgeUrl(baseUrl, subject, status, color, options) {
-  const path = [subject, status, color].map(encodeField).join('-')
-  return resolveUrl(`/badge/${path}.svg`, baseUrl, options)
+export function staticBadgeUrl(baseUrl, label, message, color, options) {
+  const path = makeStaticBadgeUrl({ label, message, color })
+  return resolveUrl(path, baseUrl, options)
```

Comment:
And delegate to the new function, while keeping the old options handling for now.